testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -0,0 +1,440 @@
+			}
+			return compareResponseObjects(castedp, castedl)
+		},
+	},
+	"/beacon/headers/{param1}": {
+		basepath: v1MiddlewarePathTemplate,
+		params: func(_ string, e types.Epoch) []string {
+			slot := uint64(0)
+			if uint64(e) > uint64(0) {
+				slot = (uint64(e) * uint64(params.BeaconConfig().SlotsPerEpoch)) - 1
+			}
+			return []string{fmt.Sprintf("%v", slot)}
+		},
+		prysmResps: map[string]interface{}{
+			"json": &apimiddleware.BlockHeaderResponseJson{},
+		},
+		lighthouseResps: map[string]interface{}{
+			"json": &apimiddleware.BlockHeaderResponseJson{},
+		},
+	},
+	"/node/identity": {
+		basepath: v1MiddlewarePathTemplate,
+		params: func(_ string, _ types.Epoch) []string {
+			return []string{"skip"}
+		},
+		prysmResps: map[string]interface{}{
+			"json": &apimiddleware.IdentityResponseJson{},
+		},
+		lighthouseResps: map[string]interface{}{
+			"json": &apimiddleware.IdentityResponseJson{},
+		},
+		customEvaluation: func(prysmResp interface{}, lhouseResp interface{}) error {
+			castedp, ok := prysmResp.(*apimiddleware.IdentityResponseJson)
+			if !ok {
+				return errors.New("failed to cast type")
+			}
+			castedl, ok := lhouseResp.(*apimiddleware.IdentityResponseJson)
+			if !ok {
+				return errors.New("failed to cast type")
+			}
+			if castedp.Data == nil {
+				return errors.New("prysm node identity was empty")
+			}
+			if castedl.Data == nil {
+				return errors.New("lighthouse node identity was empty")
+			}
+			return nil
+		},
+	},
+	"/node/peers": {
+		basepath: v1MiddlewarePathTemplate,
+		params: func(_ string, _ types.Epoch) []string {
+			return []string{"skip"}
+		},
+		prysmResps: map[string]interface{}{
+			"json": &apimiddleware.PeersResponseJson{},
+		},
+		lighthouseResps: map[string]interface{}{
+			"json": &apimiddleware.PeersResponseJson{},
+		},
+		customEvaluation: func(prysmResp interface{}, lhouseResp interface{}) error {
+			castedp, ok := prysmResp.(*apimiddleware.PeersResponseJson)
+			if !ok {
+				return errors.New("failed to cast type")
+			}
+			castedl, ok := lhouseResp.(*apimiddleware.PeersResponseJson)
+			if !ok {
+				return errors.New("failed to cast type")
+			}
+			if castedp.Data == nil {
+				return errors.New("prysm node identity was empty")
+			}
+			if castedl.Data == nil {
+				return errors.New("lighthouse node identity was empty")
+			}
+			return nil
+		},
+	},
+}
+
+func withCompareBeaconAPIs(beaconNodeIdx int, conn *grpc.ClientConn) error {
+	ctx := context.Background()
+	beaconClient := service.NewBeaconChainClient(conn)
+	genesisData, err := beaconClient.GetGenesis(ctx, &empty.Empty{})
+	if err != nil {
+		return errors.Wrap(err, "error getting genesis data")
+	}
+	currentEpoch := slots.EpochsSinceGenesis(genesisData.Data.GenesisTime.AsTime())
+
+	for path, meta := range beaconPathsAndObjects {
+		for key := range meta.prysmResps {
+			switch key {
+			case "json":
+				jsonparams := meta.params("json", currentEpoch)
+				if len(jsonparams) == 0 {
+					continue
```

Comment:
I am not exactly sure what this signifies, but shouldn't this be an error condition? Similarly for ssz.

Reply:
no this decides what params go into the url, and skips if there's nothing in the returned. it helps construct the url with some value and so I used skip but I could also use empty string for the ones that should compare some how but don't have parameter values.

Reply:
Just return `[]string{}` instead. In such case the `for` loop inside `pathFromParams` will never run.